### PR TITLE
Genesis: Add support for CrateDB

### PR DESCRIPTION
**Pull request by amotl**
## About
Provide adapter for CrateDB to support LangChain's Vector Store, Document Loader, and Conversational Memory subsystems.

## Details
This patch bundles a few others heading for upstream LangChain, but did not land.
- https://github.com/langchain-ai/langchain/pull/27710
- https://github.com/langchain-ai/langchain/pull/27711
- https://github.com/langchain-ai/langchain/pull/27712
- https://github.com/langchain-ai/langchain/pull/27713

## References
- https://github.com/crate-workbench/langchain/pull/1
- https://github.com/crate/cratedb-examples/pull/773

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2024 LangChain, Inc.
+Copyright (c) 2024 Crate.io, Inc.
```
[the remark this answers is not in the document]
Comment (amotl) [reply LICENSE:3]: langchain-cratedb is vendoring large portions of [langchain-postgres](https://github.com/langchain-ai/langchain-postgres) and [langchain-mongodb](https://github.com/langchain-ai/langchain-mongodb), mostly test cases.

> The project uses the MIT license, like the langchain-postgres project it is deriving from.
>
> -- https://github.com/crate/langchain-cratedb?tab=readme-ov-file#license

File: README.md [hunk 2]
```diff
@@ -1,45 +1,165 @@
...
+## Project Information
+
+### Acknowledgements
+Kudos to the authors of all the many software components this library is
+inheriting from and building upon, most notably the [langchain-postgres]
+package, and [langchain] itself.
+
+### Contributing
+The `langchain-cratedb` package is an open source project, and is
+[managed on GitHub]. The project is still in its infancy, and
```
[the remark this answers is not in the document]
Comment (amotl) [reply README.md]: Thanks. Fix has been included into 30ca1402df.

File: LICENSE (hunk 1 above)
[the remark this answers is not in the document]
Comment (kneth) [reply LICENSE:3]: Let's go with the MIT license

File: .github/workflows/_release.yml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+          name: dist
+          path: ${{ inputs.working-directory }}/dist/
+
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          packages-dir: ${{ inputs.working-directory }}/dist/
+          verbose: true
+          print-hash: true
+          # Temp workaround since attestations are on by default as of gh-action-pypi-publish v1\.11\.0
```
[the remark this answers is not in the document]
Comment (kneth) [reply .github/workflows/_release.yml:120]: Good to know. We should look into it but it doesn't have to be now.

**Review by kneth — APPROVED** [review]
My comments have either been addresses or answered.

**Pull request merged**